dev/breeze/src/airflow_breeze/prod/build_prod_image.py

```diff
@@ -0,0 +1,153 @@
+    "additional_python_deps",
+    "additional_dev_apt_command",
+    "additional_dev_apt_deps",
+    "additional_dev_apt_env",
+    "additional_runtime_apt_command",
+    "additional_runtime_apt_deps",
+    "additional_runtime_apt_env",
+    "upgrade_to_newer_dependencies",
+    "constraints_github_repository",
+    "airflow_constraints",
+    "airflow_image_repository",
+    "airflow_image_date_created",
+    "build_id",
+    "commit_sha",
+    "airflow_image_readme_url",
+    "install_providers_from_sources",
+    "install_from_pypi",
+    "install_from_docker_context_files",
+]
+
+PARAMS_TO_VERIFY_PROD_IMAGE = [
+    "dev_apt_command",
+    "dev_apt_deps",
+    "runtime_apt_command",
+    "runtime_apt_deps",
+]
+
+
+def construct_arguments_docker_command(prod_image: ProdParams) -> List[str]:
+    args_command = []
+    for param in PARAMS_PROD_IMAGE:
+        args_command.append("--build-arg")
+        args_command.append(param.upper() + "=" + str(getattr(prod_image, param)))
+    for verify_param in PARAMS_TO_VERIFY_PROD_IMAGE:
+        param_value = str(getattr(prod_image, verify_param))
+        if len(param_value) > 0:
+            args_command.append("--build-arg")
+            args_command.append(verify_param.upper() + "=" + param_value)
+    docker_cache = prod_image.docker_cache_prod_directive
+    if len(docker_cache) > 0:
+        args_command.extend(prod_image.docker_cache_prod_directive)
+    return args_command
+
+
+def construct_docker_command(prod_image: ProdParams) -> List[str]:
+    arguments = construct_arguments_docker_command(prod_image)
+    build_command = prod_image.check_buildx_plugin_build_command()
+    build_flags = prod_image.extra_docker_build_flags
+    final_command = []
+    final_command.extend(["docker"])
+    final_command.extend(build_command)
+    final_command.extend(build_flags)
+    final_command.extend(["--pull"])
+    final_command.extend(arguments)
+    final_command.extend(["-t", prod_image.airflow_prod_image_name, "--target", "main", "."])
+    final_command.extend(["-f", 'Dockerfile'])
+    final_command.extend(["--platform", prod_image.platform])
+    return final_command
+
+
+def login_to_docker_registry(build_params: ProdParams):
+    if build_params.ci == "true":
+        if len(build_params.github_token) == 0:
+            console.print("\n[blue]Skip logging in to GitHub Registry. No Token available!")
+        elif build_params.airflow_login_to_github_registry != "true":
+            console.print(
+                "\n[blue]Skip logging in to GitHub Registry.\
+                    AIRFLOW_LOGIN_TO_GITHUB_REGISTRY is set as false"
+            )
+        elif len(build_params.github_token) > 0:
+            run_command(['docker', 'logout', 'ghcr.io'], verbose=True, text=True)
+            # echo "${token}" | docker_v login \
+            #     --username "${GITHUB_USERNAME:-apache}" \
+            #     --password-stdin \
+            #     "ghcr.io" || true
+            run_command(
```

Comment:
We should pass the token here to login from `GITHUB_TOKEN` variable (which might be actually added as a new `--github-token` option with GITHUB_TOKEN envvar). This can be done by adding a new feature to run_command to pass. Apparently this can be easily done with the `input` parameter: 

https://stackoverflow.com/questions/163542/how-do-i-pass-a-string-into-subprocess-popen-using-the-stdin-argument